server/src/main/java/pl/edu/agh/server/security/LoginController.java

```diff
@@ -0,0 +1,54 @@
+package pl.edu.agh.server.security;
+
+import jakarta.validation.constraints.NotBlank;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+public class LoginController {
+
+  private final UserAuthorizationService userAuthorizationService;
+
+  @PostMapping("/login")
+  public Token login(@RequestBody LoginCredentials loginCredentials) {
+    return new Token(userAuthorizationService.authenticate(loginCredentials.getUsername(),
+        loginCredentials.getPassword()));
+  }
+
+  @PostMapping("/register")
+  public Token register(@RequestBody RegisterCredentials registerCredentials) {
+    return new Token(userAuthorizationService.registerUser(registerCredentials.getUsername(),
+        registerCredentials.getPassword(),
+        registerCredentials.getRepeatPassword()));
+  }
+
+  @Getter
```

Comment:
Czemu w controllerze trzymamy statyczne klasy?

Reply:
to dto dla kontrolera, sa prywatne i nie uzywane nigdzie poza pakietem